apsuite/commisslib/measure_orbit_stability.py

```diff
@@ -190,17 +200,20 @@ def calc_integrated_spectrum(self, spec, inverse=False):
             intpsd = _np.sqrt(2*_np.cumsum(spec2, axis=0))
         return intpsd
 
-    def remove_switching_freq(self, orbx=None, orby=None):
+    def filter_switching(self, orb):
         """."""
-        orbx = orbx or self.orbx.copy()
-        orby = orby or self.orby.copy()
         fs = self.sampling_freq
         fsw = self.switching_freq
-        fil_orbx, freq = self.filter_matrix(
-            orbx, fmin=0, fmax=fsw*0.9, fs=fs)
-        fil_orby, _ = self.filter_matrix(
-            orby, fmin=0, fmax=fsw*0.9, fs=fs)
-        return fil_orbx, fil_orby, freq
+        frev = self.rf_freq/self.HARM_NUM
+        if fs == frev:
```

Comment:
this is not a good way to identify TbT data. Try to do this avoiding this float comparison.
I also think this filtering algorithm is not particular to TbT data. It could also be used for ADC or FOFB rates.

Reply:
You're right. Following what we discussed, now it will filter switching only if switching mode was on during acquisition and the sampling frequency is greater than switching frequency (FOFB, ADC, TbT rates).